File: services/server/src/protocol/messages/ask_winners.py
from protocol.common.utils import bytes_to_uint32
from protocol.common import errors
from protocol.messages.bet import AGENCY_ID_LEN_BYTES

class AskWinners:
    def __init__(self, agency_id):
        self.agency_id = agency_id


    @classmethod
    def from_bytes(cls, data):
        if len(data) < AGENCY_ID_LEN_BYTES:
            raise ValueError(errors.ASK_WINNERS_TOO_SHORT_ERR)
        try:
            agency_id = bytes_to_uint32(data[:AGENCY_ID_LEN_BYTES])
            return cls(agency_id)
        except Exception as e:
            raise ValueError(f"{errors.DESERIALIZE_ASK_WINNERS_ERR}: {e}")

File: services/client/src/protocol/messages/ask_winners.go
package messages

import (
	"fmt"

	common "github.com/7574-sistemas-distribuidos/tp-nivelador/src/protocol/common"
)

type AskWinners struct {
	agencyId uint32
}

func NewAskWinners(agencyId uint32) AskWinners {
	return AskWinners{agencyId: agencyId}
}

func (a *AskWinners) ToBytes() []byte {
	bytes := make([]byte, 0, AGENCY_ID_LEN_BYTES)

	agencyIdBytes := common.Uint32ToBytes(a.agencyId)
	bytes = append(bytes, agencyIdBytes...)

	return bytes
}

func AskWinnersFromBytes(data []byte) (*AskWinners, error) {
	if len(data) < AGENCY_ID_LEN_BYTES {
		return nil, fmt.Errorf(common.AskWinnersTooShort)
	}

	agencyId, err := common.BytesToUint32(data[:AGENCY_ID_LEN_BYTES])
	if err != nil {
		return nil, fmt.Errorf(common.DeserializeAskWinnersError)
	}

	return &AskWinners{agencyId: agencyId}, nil
}
